include/dlaf/sender/transform_mpi.h

```diff
@@ -50,23 +66,25 @@ struct MPICallHelper {
   std::decay_t<F> f;
   template <typename... Ts>
   auto operator()(Ts&&... ts) -> decltype(pika::unwrapping(std::move(f))(
-      movePromiseGuard(dlaf::internal::getReferenceWrapper(ts))..., std::declval<MPI_Request*>())) {
+      getPromiseGuardRef(movePromiseGuard(dlaf::internal::getReferenceWrapper(ts)))...,
+      std::declval<MPI_Request*>())) {
     MPI_Request req;
     auto is_request_completed = [&req] {
       int flag;
       MPI_Test(&req, &flag, MPI_STATUS_IGNORE);
       return flag == 0;
     };
 
-    using result_type = decltype(pika::unwrapping(
-        std::move(f))(movePromiseGuard(dlaf::internal::getReferenceWrapper(ts))..., &req));
+    using result_type = decltype(pika::unwrapping(std::move(
+        f))(getPromiseGuardRef(movePromiseGuard(dlaf::internal::getReferenceWrapper(ts)))..., &req));
     if constexpr (std::is_void_v<result_type>) {
-      pika::unwrapping(std::move(f))(movePromiseGuard(dlaf::internal::getReferenceWrapper(ts))..., &req);
+      pika::unwrapping(std::move(
+          f))(getPromiseGuardRef(movePromiseGuard(dlaf::internal::getReferenceWrapper(ts)))..., &req);
```

Comment:
Something like this:
```suggestion
      pika::unwrapping(std::move(
          f))(getPromiseGuardRef(dlaf::internal::getReferenceWrapper(ts))..., &req);
      consumePromiseGuard(ts...);
```
where `consumePromiseGuard` would be something like this:
```
template <typename T>
void consumePromiseGuard(T& t) {
  if constexpr (IsPromiseGuardCommunicator<std::decay_t<T>>::value) {
     auto t_local = std::move(t);
  }
```

Reply:
That's a good point. I think adding `Communicator` to the name is a good idea.